### PR TITLE
Images garbage collection should be limited to the application's registry namespace

### PR DESCRIPTION
**Pull request by sgalsaleh**
#### What type of PR is this?

type::bug

#### What this PR does / why we need it:
Limits images garbage collection to the application's registry namespace

#### Which issue(s) this PR fixes:

Fixes [SC-46748](https://app.shortcut.com/replicated/story/46748/images-garbage-collection-should-be-limited-to-the-application-s-registry-namespace)

#### Does this PR introduce a user-facing change?
```release-note
Fixes an issue that causes images garbage collection in embedded clusters to remove images outside of the application's dedicated registry namespace.
```

#### Does this PR require documentation?
NONE

### CODE REVIEW

File: pkg/registry/images.go [hunk 1]
```diff
@@ -215,6 +215,17 @@ func deleteUnusedImages(ctx context.Context, registry types.RegistrySettings, us
 	digestsInRegistry := map[string]string{}
 	for _, r := range searchResult {
 		imageName := path.Join(registry.Hostname, r.Name)
+
+		parts := strings.Split(imageName, "/")
```
Comment (emosbaugh) [inline pkg/registry/images.go:219]: Why not use `reference.Path(ref.DockerReference())`?

**Review by emosbaugh — APPROVED** [review]
This looks correct so i will approve but why not use docker libs they provide?

File: pkg/registry/images.go (hunk 1 above)
Comment (sgalsaleh) [reply pkg/registry/images.go:219]: cuz it's not parsed as a reference yet at that point. i guess i could, but we do use this pattern in our codebase

**Comment by emosbaugh** [comment]
It is parsed on the next line

Sent via Superhuman iOS ***@***.***>

**Review by emosbaugh — APPROVED** [review]

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**